clang/include/clang/Basic/Attr.td

```diff
@@ -1313,6 +1313,40 @@ def SYCLIntelNoGlobalWorkOffset : InheritableAttr {
   let PragmaAttributeSupport = 0;
 }
 
+def SYCLIntelLoopFuse : InheritableAttr {
+  let Spellings = [CXX11<"intel","loop_fuse">];
+  let Args = [ExprArgument<"Value", /*optional=*/ 1>];
+  let LangOpts = [SYCLIsDevice, SYCLIsHost];
+  let Subjects = SubjectList<[Function], ErrorDiag>;
+  let Documentation = [SYCLIntelLoopFuseDocs];
+  let PragmaAttributeSupport = 0;
```

Comment:
@MrSidims , could you comment on this? Please correct me if I'm wrong, but I remember that you disabled pragma attribute support for FPGA attributes for some reason.

Reply:
This call was made, because we don’t expect/don’t want people to start using pragma spellings, we want them to exclusively use the attribute spellings, see: https://github.com/intel/llvm/issues/1813 and https://github.com/intel/llvm/pull/1815

Reply:
It's not about FPGa attributes only, but about SYCL attributes in general.

Reply:
Huh, #1813 is missing an explanation for why this is desired. @erichkeane, can you provide context?

FWIW, the typical upstream expectation is that new attributes are supported via #pragma clang attribute (https://clang.llvm.org/docs/LanguageExtensions.html#specifying-an-attribute-for-multiple-declarations-pragma-clang-attribute) unless there's some reason for them not to (like statement or type attributes). This is a convenience pragma and it'd be a bit strange to exclude our declaration attributes as a matter of preference rather than on some technical grounds.

Reply:
The reasoning in SYCL is that we want to encourage the attribute spellings as we want sycl programs to be as close to 'pure C++' as possible.  

That said, that #pragma clang is REALLY useful looking.  I was actually unaware of it when I made that comment.  Is there a way to opt these into THAT without opting into #pragma AttrName (which I think was my original thoguht?)?  If so, I think doing THAT shows value here.

Reply:
> The reasoning in SYCL is that we want to encourage the attribute spellings as we want sycl programs to be as close to 'pure C++' as possible.

Pffft, as if `#pragma` isn't pure C++. Look, it's right here in the standard: http://eel.is/c++draft/cpp.pragma and even has the stable name `cpp` (can you get more pure C++ than that?)! ;-)

> That said, that #pragma clang is REALLY useful looking. I was actually unaware of it when I made that comment. Is there a way to opt these into THAT without opting into #pragma AttrName (which I think was my original thoguht?)? If so, I think doing THAT shows value here.

All declaration attributes with "normal" subjects (like the ones in this patch) are automatically opted into that pragma support unless they specify `PragmaAttributeSupport = 0` as in this patch. The only way you get `#pragma AttrName` style pragmas is to use an explicit `Pragma` spelling, as in the `InitSeg` or `LoopHint` attributes. So I think what we really want to do is strip out all of the `PragmaAttributeSupport = 0` work from #1815 and friends.

Reply:
Ah!  Well then! I would be in favor of said patch.

Reply:
Also:

>>Pffft, as if #pragma isn't pure C++. Look, it's right here in the standard: http://eel.is/c++draft/cpp.pragma and even has the stable name cpp (can you get more pure C++ than that?)! ;-)

Them's fighten words in these parts.

Reply:
LOL, well then I'll pick my battles better than to fight over whether `#pragma` is pure C++ given that we could fight over much more entertaining things, like whether `any_invokable` is a good name. ;-)

Reply:
Even if I dislike the concept of attributes and `#pragma` in general, I can already hear some customers asking for some compatibility with their HLS source code they want to reuse with a lot of `#pragma` extensions.

So probably we will have some SYCL compilers with some `#pragma` vendor extensions...

Reply:
I've enabled it for this attribute. I'll correct the other attributes in a separate patch.

Reply:
Github issue to enable PragmaAttributeSupport for SYCL attributes filed - https://github.com/intel/llvm/issues/2910